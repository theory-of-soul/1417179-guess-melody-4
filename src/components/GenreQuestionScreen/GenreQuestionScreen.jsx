import React from "react";
import PropTypes from "prop-types";

const GenreQuestionScreen = (props) => {
  const {
    question,
    handleAnswer,
    renderAudioPlayer,
    answers: userAnswers,
    onChooseAnswer
  } = props;

  return (
    <React.Fragment>
      <h2 className="game__title">Выберите {question.genre} треки</h2>
      <form
        className="game__tracks"
        onSubmit={(event) => {
          event.preventDefault();
          handleAnswer(question, userAnswers);
        }}
      >
        {
          question.answers.map((answer, index) => {
            return (
              <div className="track" key={`${answer.genre}-${index}`}>
                {
                  renderAudioPlayer(answer.audioSrc, index)
                }
                <div className="game__answer">
                  <input
                    className="game__input visually-hidden"
                    type="checkbox"
                    name="answer"
                    value={`answer-${index}`}
                    id={`answer-${index}`}
                    checked={userAnswers[index]}
                    onChange={() => onChooseAnswer(index)}
                  />
                  <label className="game__check" htmlFor={`answer-${index}`}>Отметить</label>
                </div>
              </div>
            );
          })
        }

        <button className="game__submit button" type="submit">Ответить</button>
      </form>
    </React.Fragment>
  );
};

GenreQuestionScreen.propTypes = {
  question: PropTypes.shape({
    type: PropTypes.string.isRequired,
    genre: PropTypes.string.isRequired,
    answers: PropTypes.arrayOf(
        PropTypes.shape({
          genre: PropTypes.string.isRequired,
          audioSrc: PropTypes.string.isRequired,
        })
    ).isRequired
  }),
  handleAnswer: PropTypes.func.isRequired,
  renderAudioPlayer: PropTypes.func.isRequired,
  answers: PropTypes.shape({
    0: PropTypes.bool,
    1: PropTypes.bool,
    2: PropTypes.bool,
    3: PropTypes.bool
  }).isRequired,
  onChooseAnswer: PropTypes.func.isRequired
};

export default React.memo(GenreQuestionScreen);
