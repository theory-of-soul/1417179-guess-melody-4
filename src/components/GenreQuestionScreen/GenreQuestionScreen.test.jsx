import React from "react";
import renderer from 'react-test-renderer';
import GenreQuestionScreen from "./GenreQuestionScreen";

const question = {
  type: `genre`,
  genre: `инди-рок`,
  answers: [{
    genre: `инди-рок`,
  }, {
    genre: `рок`,
  }, {
    genre: `поп`,
  }, {
    genre: `рок`,
  }]
};

describe(`GenreQuestionScreen snapshot tests`, () => {
  it(`GenreQuestionScreen should render 4 melody`, () => {
    const component = renderer
      .create(<GenreQuestionScreen question={question} handleAnswer={() => {}} />)
      .toJSON();

    expect(component).toMatchSnapshot();
  });
});
