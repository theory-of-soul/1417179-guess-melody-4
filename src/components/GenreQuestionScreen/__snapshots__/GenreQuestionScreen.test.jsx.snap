// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenreQuestionScreen snapshot tests GenreQuestionScreen should render 4 melody 1`] = `
<section
  className="game game--genre"
>
  <header
    className="game__header"
  >
    <a
      className="game__back"
      href="#"
    >
      <span
        className="visually-hidden"
      >
        Сыграть ещё раз
      </span>
      <img
        alt="Угадай мелодию"
        className="game__logo"
        src="img/melody-logo-ginger.png"
      />
    </a>
    <div
      className="game__mistakes"
    >
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
    </div>
  </header>
  <section
    className="game__screen"
  >
    <h2
      className="game__title"
    >
      Выберите 
      инди-рок
       треки
    </h2>
    <form
      className="game__tracks"
    >
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          type="button"
        />
        <div
          className="track__status"
        >
          <audio />
        </div>
        <div
          className="game__answer"
        >
          <input
            className="game__input visually-hidden"
            id="answer-0"
            name="answer"
            type="checkbox"
            value="answer-0"
          />
          <label
            className="game__check"
            htmlFor="answer-1"
          >
            Отметить
          </label>
        </div>
      </div>
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          type="button"
        />
        <div
          className="track__status"
        >
          <audio />
        </div>
        <div
          className="game__answer"
        >
          <input
            className="game__input visually-hidden"
            id="answer-1"
            name="answer"
            type="checkbox"
            value="answer-1"
          />
          <label
            className="game__check"
            htmlFor="answer-1"
          >
            Отметить
          </label>
        </div>
      </div>
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          type="button"
        />
        <div
          className="track__status"
        >
          <audio />
        </div>
        <div
          className="game__answer"
        >
          <input
            className="game__input visually-hidden"
            id="answer-2"
            name="answer"
            type="checkbox"
            value="answer-2"
          />
          <label
            className="game__check"
            htmlFor="answer-1"
          >
            Отметить
          </label>
        </div>
      </div>
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          type="button"
        />
        <div
          className="track__status"
        >
          <audio />
        </div>
        <div
          className="game__answer"
        >
          <input
            className="game__input visually-hidden"
            id="answer-3"
            name="answer"
            type="checkbox"
            value="answer-3"
          />
          <label
            className="game__check"
            htmlFor="answer-1"
          >
            Отметить
          </label>
        </div>
      </div>
      <button
        className="game__submit button"
        type="submit"
      >
        Ответить
      </button>
    </form>
  </section>
</section>
`;
